imap_upstream.adoc

```diff
@@ -126,7 +126,7 @@ connections of existing IMAP clients.
 Prior to changing any configuration, it should be decided how existing
 routes in `memcached` will be handled. By default, routes are cached in
 `memcached` for 1 day, as specified by the
-`zimbraMemcachedClientExpirySeconds` LDAP attribute. If untouched, it
+`zimbraReverseProxyCacheEntryTTL` LDAP attribute. If untouched, it
```

Comment:
@gordyt you wrote the original?  The memcached which uses `zimbraMemcachedClientExpirySeconds` is used by callers of `MemcachedConnector.getClient()`

```
getClient() : ZimbraMemcachedClient - com.zimbra.cs.memcached.MemcachedConnector
	CalListCache() - com.zimbra.cs.mailbox.calendar.cache.CalListCache
	CalSummaryMemcachedCache() - com.zimbra.cs.mailbox.calendar.cache.CalSummaryMemcachedCache
	CtagInfoCache() - com.zimbra.cs.mailbox.calendar.cache.CtagInfoCache
	CtagResponseCache() - com.zimbra.cs.mailbox.calendar.cache.CtagResponseCache
	EffectiveACLCache() - com.zimbra.cs.mailbox.acl.EffectiveACLCache
	FoldersTagsCache() - com.zimbra.cs.mailbox.FoldersTagsCache
	handle(Element, Map<String, Object>) : Element - com.zimbra.cs.service.admin.GetMemcachedClientConfig
	map - com.zimbra.cs.imap.MemcachedImapCache
	MemcachedItemCache() - com.zimbra.cs.mailbox.MemcachedItemCache
	OAuthTokenCache() - com.zimbra.cs.account.oauth.OAuthTokenCache
```
Perhaps `com.zimbra.cs.imap.MemcachedImapCache` is relevant?? 
If it turns out `zimbraMemcachedClientExpirySeconds` is wrong, then the suggestions below about running `zmprov rmcc all` are presumably wrong as well.

Reply:
@fatkudu `zimbraReverseProxyCacheEntryTTL` is the correct attribute.  I just checked in `ProxyConfGen` and it is using `zimbraReverseProxyCacheEntryTTL` to set the value of `memcache.ttl`.  That value is used in `nginx.conf.memcache.template` to specify the `ttl` value in the `memcached` config block.

Reply:
...and yes this block should be updated as well (replacing `zimbraMemcachedClientExpirySeconds` with `zimbraReverseProxyCacheEntryTTL`:

> One day prior to switching to the imapd pool, modify the value zimbraMemcachedClientExpirySeconds of to a shorter interval, such as 30 minutes, and reload the memcached configuration with the command zmprov rmcc all. Doing so will allow embedded IMAP to be turned off within a shorter timeframe without risking service interruptions

Reply:
@gordyt - how will `zmprov rmcc all` make things better if `zimbraReverseProxyCacheEntryTTL` is changed?  I'm suspecting the proxies either need restarting or some kind of reload.

Reply:
@fatkudu - `zimbraReverseProxyCacheEntryTTL` is one of the attributes that are monitored by `zmconfigd`, so any changes to it should automatically restart the proxy servers.

Reply:
Sounds good.  Then we just need to remove the redundant text about running `zmprov rmcc all` which is unnecessary as it would affect `zimbraMemcachedClientExpirySeconds`, which isn't relevant but does not affect `zimbraReverseProxyCacheEntryTTL` which is.

Reply:
Thank you for your reviewing. I will remove that part from ", and reload the memcached" to the end of the paragraph.